pkg/cmd/deadman/operator_condition_challenger.go

```diff
@@ -0,0 +1,179 @@
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+)
+
+type OperatorConditionChallenger struct {
+	// operatorToTimeToCheckForStaleness is a map from operator name to the time at which the condition messages will be
+	// reset to a "checking for activity" message.
+	operatorToTimeToCheckForStaleness map[string]time.Time
+
+	durationAllowedBetweenStalenessChecks time.Duration
+
+	clusterOperatorClient configv1versionedtypes.ClusterOperatorInterface
+	clusterOperatorLister configv1lister.ClusterOperatorLister
+
+	queue workqueue.RateLimitingInterface
+}
+
+// NewOperatorConditionChallenger is a controller that sets operator condition messages to ensure that the operator
+// in question updates the condition messages back.  If the condition messages are not changed, then the operator
+// condition is considered stale.
+func NewOperatorConditionChallenger(
+	configClient configeversionedclient.Interface,
+	configInformers configv1informers.SharedInformerFactory,
+	durationAllowedBetweenStalenessChecks time.Duration,
+) *OperatorConditionChallenger {
+
+	c := &OperatorConditionChallenger{
+		operatorToTimeToCheckForStaleness:     map[string]time.Time{},
+		durationAllowedBetweenStalenessChecks: durationAllowedBetweenStalenessChecks,
+
+		clusterOperatorClient: configClient.ConfigV1().ClusterOperators(),
+		clusterOperatorLister: configInformers.Config().V1().ClusterOperators().Lister(),
+
+		queue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "OperatorConditionChallenger"),
+	}
+
+	configInformers.Config().V1().ClusterOperators().Informer().AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				clusterOperator := obj.(*configv1.ClusterOperator)
+				c.queue.Add(clusterOperator.Name)
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				clusterOperator := newObj.(*configv1.ClusterOperator)
+				c.queue.Add(clusterOperator.Name)
+			},
+		},
+		1*time.Minute,
+	)
+
+	return c
+}
+
+func getMostRecentConditionTransitionTime(clusterOperator *configv1.ClusterOperator) *time.Time {
+	if len(clusterOperator.Status.Conditions) == 0 {
+		return nil
+	}
+
+	newestTime := time.Time{}
+	for _, condition := range clusterOperator.Status.Conditions {
+		if condition.LastTransitionTime.After(newestTime) {
+			newestTime = condition.LastTransitionTime.Time
+		}
+	}
+	return &newestTime
+}
+
+func (c *OperatorConditionChallenger) syncHandler(ctx context.Context, key string) error {
+	clusterOperator, err := c.clusterOperatorLister.Get(key)
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	if isCheckingForStaleness(clusterOperator) {
+		// staleness check is in progress already, just return
+		return nil
+	}
+
+	mostRecentOperatorUpdate := getMostRecentConditionTransitionTime(clusterOperator)
+	if mostRecentOperatorUpdate == nil {
+		// this means we have no conditions, so there's no work to do.
```

Comment:
Is it? Do we want to cover a case where the operator never updates its status, so the condition is stale/empty? Or a chance when third-party actors erase conditions and the operator never sets then back?

Reply:
> Is it? Do we want to cover a case where the operator never updates its status, so the condition is stale/empty? Or a chance when third-party actors erase conditions and the operator never sets then back?

If we have no conditions, it's not clear what this controller should set.  I think this logic would only fail during failed installations.